onnxscript/ir/_core.py

```diff
@@ -1254,61 +1259,176 @@ def metadata_props(self) -> dict[str, str]:
+        + textwrap.indent("\n".join(node_lines), " " * 4)
+        + textwrap.indent(f"\nreturn {returns}", " " * 4)
+        + "\n}"
+    )
 
-        return f"{signature} {body}"
+    return f"{signature} {body}"
 
-    def __repr__(self) -> str:
-        inputs_text = "\n" + ",\n".join(str(x) for x in self.inputs)
-        outputs_text = "\n" + ",\n".join(str(x) for x in self.outputs)
-        initializers_text = ",\n".join(str(x) for x in self.initializers.values())
-        if initializers_text:
-            initializers_text = (
-                "\ninitializers=(\n" + textwrap.indent(initializers_text, " " * 4) + "\n),"
-            )
-        return f"""\
-{self.__class__.__name__}(
-    name={self.name or ':anonymous_graph:' + str(id(self))!r},
+
+def _graph_repr(graph: Graph | GraphView) -> str:
+    """Return an repr string of the graph."""
+    inputs_text = "\n" + ",\n".join(str(x) for x in graph.inputs)
+    outputs_text = "\n" + ",\n".join(str(x) for x in graph.outputs)
+    initializers_text = ",\n".join(str(x) for x in graph.initializers.values())
+    if initializers_text:
+        initializers_text = (
+            "\ninitializers=(\n" + textwrap.indent(initializers_text, " " * 4) + "\n),"
+        )
+    return f"""\
+{graph.__class__.__name__}(
+    name={graph.name or 'anonymous_graph:' + str(id(graph))!r},
     inputs=({textwrap.indent(inputs_text, ' '*8)}
     ),
     outputs=({textwrap.indent(outputs_text, ' '*8)}
     ),{textwrap.indent(initializers_text, ' '*4)}
-    len(nodes)={len(self._nodes)}
+    len()={len(graph)}
 )"""
 
 
+class GraphView(Sequence[Node], _display.PrettyPrintable):
+    """A read-only view on a graph.
+
+    The GraphView is useful for analysis of a subgraph. It can be initialized
+    with a subset of nodes from a :class:`Graph`. Creating GraphView does not
+    change the ownership of the nodes, and so it is possible to create multiple
+    GraphViews that contain the same nodes. If the underlying nodes / connections
+    are mutated, the mutation will be reflected in all views as well.
+
+    The graph view can be serialized to ONNX::
+
+            graph_proto = ir.serde.serialize_graph(graph_view)
+
+    It can also be used to create a model::
+
+            model = ir.Model(graph_view, ir_version=8)
+            model_proto = ir.serde.serialize_model(model)
+            
+    The model created with a GraphView will have a fixed topology, and its graph
+    will remain read-only as a GraphView. No copying will be done during the
+    initialization process.
+
+    Attributes:
+        name: The name of the graph.
+        inputs: The input values of the graph.
+        outputs: The output values of the graph.
+        nodes: All nodes visible in this view. They do not have to be sorted.
+        initializers: The initializers in the graph.
+        doc_string: Documentation string.
+        opset_imports: Opsets imported by the graph.
+        metadata_props: Metadata.
+    """
```

Comment:
That would be great to add an example of the outputs in the documentation to understand what it looks like.

Reply:
Do you mean what the object prints out, or the graph outputs?

Reply:
I think I understand now: The GraphView is not just meant to be displayed. I touch the repr methods because I wanted to share them between Graph and GraphView. I will add an example in the repr function later.